### PR TITLE
feat: [HL-20] Create sign button with social media view 📖

### PR DESCRIPTION
**Pull request by abdallah-7698**
create reusable View contains sign button and social media buttons

### CODE REVIEW

File: Healthy/Classes/Generated/UIImage.Generated.swift [hunk 1]
```diff
@@ -9,4 +9,12 @@ extension UIImage {
     static var accentColor: UIImage {
       UIImage(named: "AccentColor")!
     }
+
+    static var facebookIcon: UIImage {
+      UIImage(named: "facebook_icon")!
+    }
+
+    static var googleIcon: UIImage {
+      UIImage(named: "google_icon")!
```
Comment (ahmdmhasn) [inline Healthy/Classes/Generated/UIImage.Generated.swift]:
```suggestion
      UIImage(named: "icon-facebook")!
    }

    static var googleIcon: UIImage {
      UIImage(named: "icon-google")!
```

Please rename the assets to match the given name.

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import UIKit
+
+class SignButtonWithSocialMediaView: UIView {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]:
```suggestion
final class SignButtonWithSocialMediaView: UIView {
```

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        mediaBackgroundViews.forEach { $0.setupDefaultShadow()}
+    }
+}
+
+// MARK: - configure
+
+extension SignButtonWithSocialMediaView {
+    enum SignButtonTitle: String {
+        case signin = "Sign In"
+        case signup = "Sign Up"
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]: Please use auto-generated localizable strings.

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    func configure(buttonTitle: SignButtonTitle) {
+        signinButton.setTitle(buttonTitle.rawValue, for: .normal)
+    }
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]: Will not be needed based on the above comment.

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+class SignButtonWithSocialMediaView: UIView {
+
+    // MARK: - IBOutlet
+
+    @IBOutlet weak private(set) var contentView: UIView!
+    @IBOutlet weak private(set) var signinButton: UIButton!
+    @IBOutlet weak private(set) var googleButton: UIButton!
+    @IBOutlet weak private(set) var facebookButton: UIButton!
+    @IBOutlet private(set) var mediaBackgroundViews: [UIView]!
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]:
```suggestion
    @IBOutlet weak private var contentView: UIView!
    @IBOutlet weak private var signinButton: UIButton!
    @IBOutlet weak private var googleButton: UIButton!
    @IBOutlet weak private var facebookButton: UIButton!
    @IBOutlet private var mediaBackgroundViews: [UIView]!
```

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift (hunk 2 above)
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]:
```suggestion
import UIKit

struct SignButtonWithSocialMediaViewModel {
    ...
}

class SignButtonWithSocialMediaView: UIView {
```

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    }
+}
+
+// MARK: - UIView fill to Edge
+
+private extension UIView {
+    func fillSuperview(padding: UIEdgeInsets = .zero) {
+        translatesAutoresizingMaskIntoConstraints = false
+
+        guard let superview = superview else {return}
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]:
```suggestion
        guard let superview else { 
            assertionFailure("Unable to find superView")
            return
        }
        
        translatesAutoresizingMaskIntoConstraints = false
```

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    static var reusableIdentifier: String {
+        String("\(Self.self)")
+    }
+
+    func loadNibView() {
+        Bundle.main.loadNibNamed(Self.reusableIdentifier, owner: self)
+    }
+
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]: Please move to a a new file `UIView+Helpers`.swift in `Extensions`

File: Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        layer.shadowColor = UIColor.black.cgColor
+        layer.shadowOffset = CGSize(width: 0, height: 0)
+        layer.shadowOpacity = 0.1
+        layer.shadowRadius = 6.0
+    }
+}
+
+// MARK: - UIView fill to Edge
+
+private extension UIView {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/SignButtonWithSocialMediaView/SignButtonWithSocialMediaView.swift]: Please move to a a new file `UIView+Helpers.swift` in Extensions

**Review by ahmdmhasn — APPROVED** [review]

**Comment by ahmdmhasn** [comment]
I've renamed the pull request to `feat: [HL-20] Create sign button with social media view 📖`
Feat --> feat
create --> Create

[HL-20]: https://motoon-eg.atlassian.net/browse/HL-20?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by ahmdmhasn** [comment]
Neat & Organized!

**Review by abdelhamid-f-nasser — DISMISSED** [review]
Great job !! 👍🏼 🚀

**Review by djayyab — APPROVED** [review]

**Pull request merged**